### DISCUSSION TITLE
why are there redundant files in node_module?

### DISCUSSION THREAD

**Discussion by DLillard0** [pnpm/pnpm#4194] [Q&A] [answered]
### I try to add only one dependency to the project:
`  "devDependencies": {
    "eslint": "^6.7.2"
  },`

### when I run pnpm install, the node_modules has four symlink:

`eslint
eslint-scope
eslint-utils
eslint-visitor-keys
`
### I think there should be only one symlink

**Comment by zkochan** [comment, marked as answer]
This happens due to the default value of the [public-hoist-pattern](https://pnpm.io/npmrc#public-hoist-pattern) setting.

**Reply by DLillard0** [reply]
thank you, I have another question, when i do same thing in a pnpm workspace project, the package node_moudle only have eslint, that means the public-hoist-pattern setting is invalid in workspace ?

**Reply by zkochan** [reply]
in a workspace, it will be hoisted to the node_modules that is in the same directory, where `pnpm-workspace.yaml` is

**Reply by DLillard0** [reply]
I add a vue peoject generated by vue cli in the workspace, then I install and run, but it throw a err:
`Failed to resolve loader: babel-loader
You may need to install it.`
Am I missing any configuration?
It works fine when it is not in the workspace.
